include/flox/registry.hh

```diff
@@ -148,6 +160,31 @@ struct RegistryInput : public InputPreferences
     return static_cast<nix::ref<nix::FlakeRef>>( this->from );
   };
 
+  [[nodiscard]] bool
+  operator==( const RegistryInput & other ) const
+  {
+    if ( static_cast<InputPreferences>( *this )
+         != static_cast<InputPreferences>( other ) )
+      {
+        return false;
+      }
+
+    if ( this->from == other.from ) { return true; }
+
+    if ( ( this->from == nullptr ) || ( other.from == nullptr ) )
+      {
+        return false;
+      }
+
+    return this->from->to_string() != other.from->to_string();
```

Comment:
Nonblocking
```suggestion
    return *this->from == *other.from;
```

Reply:
Did that compile for you? I wasn't sure if they implemented `==`

Reply:
Yeah it did